### PR TITLE
feat: Ensure accurate read/write classification for kubectl sub commands like `kubectl rollout history`

### PR DESCRIPTION
**Pull request by yankay**
Fix: #393

Previously, all rollout commands were incorrectly categorized as write operations,
resulting in unnecessary approval prompts for read-only subcommands such as history and status.

This PR introduces dedicated mappings for subcommands like `rollout` :
It can also be applied to commands like:
- auth can-i (read-only, checks permissions)
- auth reconcile (write, updates permission rules)

### CODE REVIEW

**Comment by zvdy** [comment]
Looks good to me, perhaps add more tests, I assume you have tested manually, If not, please do so by inducing the LLM to these edge cases. Once done we can merge @droot

File: pkg/tools/kubectl_filter.go [hunk 1]
```diff
@@ -145,21 +161,26 @@ func analyzeCall(call *syntax.CallExpr) string {
 	}
 
 	verb := args[verbPos]
+	subVerb := ""
+	if verbPos+1 < len(args) && !strings.HasPrefix(args[verbPos+1], "-") {
+		subVerb = args[verbPos+1]
```
Comment (droot) [inline pkg/tools/kubectl_filter.go]: Not sure how likely it is but technically possible to specify the flags between `verb` and `subverb` :) `k --f1 rollout --f2 history ` and this check will miss it.

a few options:

1. we can extract verb and subverb in a single for loop (above) (we stop at verb today)
2. filter out flags from the args to create nonFlagArgs and then simply extract verb, subverb from the filtered args.

File: pkg/tools/kubectl_filter_test.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ func TestKubectlModifiesResource(t *testing.T) {
 			{"Logs with follow", "kubectl logs nginx -f", "no"},
 			{"Watch pods", "kubectl get pods --watch", "no"},
 			{"Watch pods short", "kubectl get pods -w", "no"},
+			{"Rollout status", "kubectl rollout status deployment nginx", "no"},
```
Comment (droot) [inline pkg/tools/kubectl_filter_test.go:50]: would be good to add flags `k --f1 rollouts --f2 status --f3 deployment nginx`

Comment (yankay) [reply pkg/tools/kubectl_filter_test.go:50]: Thanks @droot 

It has been changed :-)

I also noticed some edge cases, such as commands like `kubectl -n default get deployment` where the first non-flag argument isn't the verb.

However, I found that LLMs don't generate such commands, so this should be a rare case.
I'll leave this out of this PR for now.

**Comment by yankay** [comment]
> Looks good to me, perhaps add more tests, I assume you have tested manually, If not, please do so by inducing the LLM to these edge cases. Once done we can merge @droot

I have tested it : 

1. The `rollout status`:
img_v3_02o1_38968372-c7b8-46c2-89ae-cbe930ae762g

2. The `rollout undo`
img_v3_02o1_bbe35e48-c678-44c6-b3e4-a57f450e37bg

File: pkg/tools/kubectl_filter.go [hunk 3]
```diff
@@ -133,33 +149,39 @@ func analyzeCall(call *syntax.CallExpr) string {
...
 	klog.V(2).Infof("analyzeCall: found kubectl: %q", firstArg)
 
-	// Get the verb (first non-flag argument after kubectl)
-	verbPos := 1 // Start after kubectl at position 0
-	for verbPos < len(args) && strings.HasPrefix(args[verbPos], "-") {
-		verbPos++
+	// Find the verb: first non-flag argument after "kubectl"
+	verb := ""
+	subVerb := ""
+	for i := 1; i < len(args); i++ {
```
Comment (droot) [inline pkg/tools/kubectl_filter.go]: nit: you can simply use range here because indices are no longer useful in this case :)

```Go

for _, arg := range args {

}

```

File: pkg/tools/kubectl_filter.go [hunk 4]
```diff
@@ -133,33 +149,39 @@ func analyzeCall(call *syntax.CallExpr) string {
...
 	}
-
-	if verbPos >= len(args) {
+	if verb == "" {
 		klog.Warningf("analyzeCall: no verb found after kubectl in args: %v", args)
 		return "unknown"
 	}
 
-	verb := args[verbPos]
 	hasDryRun := hasDryRunFlag(strings.Join(args, " "))
```
Comment (droot) [inline pkg/tools/kubectl_filter.go]: couldn't resist :)

we also can check for dry-run flags in the same loop all the way and can kill this helper function :)

File: pkg/tools/kubectl_filter_test.go (hunk 2 above)
Comment (droot) [reply pkg/tools/kubectl_filter_test.go:50]: Aah... good catch. I think this may bite us down the line. Do you mind filing an issue for us to track this so that we can look into it.

**Review by droot — APPROVED** [review]
Thanks @yankay for working on this and thanks @zvdy for the review.

I have minor comments. None of them are blockers so approving the PR. If you plan to address them, let me know otherwise I will go ahead with the merge in the morning tomorrow.

File: pkg/tools/kubectl_filter.go (hunk 4 above)
Comment (yankay) [reply pkg/tools/kubectl_filter.go]: Thanks @droot 

I merged the `verbs` and  `hasDryRun`  into a single loop and was planning to delete this function.

But I noticed this function has a test case with some good cases in it.
So I refactored the loop out and used this test to verify it. What do you think?

**Review by droot — APPROVED** [review]
